BackendTaskPilot/Program.cs

```diff
@@ -0,0 +1,38 @@
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+var builder = WebApplication.CreateBuilder(args);
+var app = builder.Build();
+
+ExchangeRateService service = new ExchangeRateService();
+
+// REST API to get exchange rates based on input currencies
+app.MapPost("/", service.GetExchangeRates);
+
+app.Run();
+
+public struct CnbRatesResponse
+{
+    [JsonPropertyName("rates")] public List<Rate> Rates { get; set; }
+}
+
+public struct Rate
+{
+    [JsonPropertyName("currencyCode")] public string CurrencyCode { get; set; }
+    [JsonPropertyName("rate")] public float RateValue { get; set; }
+}
+
+internal class ExchangeRateService
+{
+    private const string MainCurrenciesUrl = "https://api.cnb.cz/cnbapi/exrates/daily";
+
+    public IEnumerable<Rate> GetExchangeRates(List<string> currencies)
+    {
+        HttpClient client = new HttpClient();
+			
+        var data = client.GetStringAsync(MainCurrenciesUrl).Result;
+        var response = JsonSerializer.Deserialize<CnbRatesResponse>(data);
+
```

Comment:
We should validate the `currencies` list against a predefined set of valid values to avoid errors and inconsistencies. Using an `enum` for supported currencies is an idiomatic way to handle this in C#.

Here’s how we could structure it:

```csharp
public enum SupportedCurrency
{
    USD,
    EUR,
    JPY,
    SAR,
    KES
}
```
Enums in C# are really great for these sorts of defined, smaller lists that you can validate against for example.

Then, validation becomes simple and readable:

```C#
private bool AreCurrenciesValid(List<string> requestedCurrencies)
{
    return requestedCurrencies != null &&
           requestedCurrencies.All(requestedCurrency => Enum.TryParse(typeof(SupportedCurrency), requestedCurrency, true, out _));
}
```

This ensures:
- **Null or empty lists** are handled upfront.
- Every requested currency is validated against the enum.

Finally, in the calling code, you could try something like this, or if we don't want exceptions, just log it to ILogger:

```C#
if (!AreCurrenciesValid(currencies))
{
    throw new ArgumentException("One or more currencies are invalid.");
}
```

This approach keeps the validation logic clean and idiomatic while leveraging the enum for safety and maintainability. There are multiple ways to approach this though, I would recommend a pair-programming session on validation logic if you are having trouble grasping any of these concepts!